### ISSUE TITLE
Periodic background tasks | Refreshing notice board entries and notifications

### ISSUE THREAD

**Issue by bennyboer** [Fachschaft07/Guide7#34]
In the old app we had a periodic pull for notice board entries and other resource data in order to check whether it changed.

People like me do not want to check the app and manually refresh the notice board everyday but rather be notified by the app about new notice board entries.

We will need a general way to support periodic background tasks. I did some research:
Currently Flutter does not directly support periodic background tasks but there are already plans to implement it in the future (See [here](https://github.com/flutter/flutter/issues/24278)). There is a plugin available which will do the job for android and iOS just fine: https://pub.dartlang.org/packages/background_fetch

The job is to create an abstraction around the used `background_fetch` plugin in order to replace it with the official Flutter implementation once it is ready. Create an example for fetching notice board entries every **half an hour**. Document the usage in the Wiki for developers to use it with ease.

### Summing up:

- Implement `background_fetch` in an own periodic background task abstraction
- Create notice board entry refresh every **30 minutes**
- Document usage in the Wiki
- Notification in case the notice board received new entries

**Comment by bennyboer** [comment]
The wiki has two new pages: `Background tasks` and `Fire Notifications`.

**Comment by bennyboer** [comment]
Implementation finished, just testing now.
